### PR TITLE
Avoid all evaluation of LazyList#foldRightDefer

### PR DESCRIPTION
**Pull request by takayahilton**

### CODE REVIEW

**Comment by travisbrown** [comment]
One of the CI failures is just a connectivity thing and the other seems definitely unrelated (see #3568), so I just restarted both jobs.

File: tests/src/test/scala-2.13+/cats/tests/LazyListSuite.scala [hunk 1]
```diff
@@ -52,6 +55,16 @@ class LazyListSuite extends CatsSuite {
     assert(LazyList.empty[Int].show === (s"LazyList()"))
   }
 
+  test("Avoid all evaluation of LazyList#foldRightDefer") {
```
Comment (travisbrown) [inline tests/src/test/scala-2.13+/cats/tests/LazyListSuite.scala:58]: Just to confirm: this test hangs without the change above?

Comment (takayahilton) [reply tests/src/test/scala-2.13+/cats/tests/LazyListSuite.scala:58]: Yes,  The test never finishes.

**Comment by travisbrown** [comment]
One Scala.js build is still failing even after a couple of restarts. It couldn't be related, so I think we can just wait and try again in a couple of hours or so.

**Comment by barambani** [comment]
It seems it was a corrupted cache. I deleted the PR's cache and it went through.

**Pull request merged**